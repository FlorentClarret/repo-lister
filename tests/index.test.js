import MockDate from 'mockdate'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import App from '../pages/index.js'

describe('With Enzyme', () => {
  it('App shows "List of posts" heading', () => {
    const app = shallow(<App posts={[{id: 1}]}/>)

    expect(app.find('h2').length).toEqual(1)
  })
})

describe('With Snapshot Testing', () => {
  beforeEach(() => MockDate.set('1/1/2000'))

  it('App matches snapshot', () => {
    const component = renderer.create(<App posts={[{id: 1}]}/>)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  afterEach(() => MockDate.reset())
})
