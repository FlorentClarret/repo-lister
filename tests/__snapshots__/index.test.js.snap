// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App matches snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-md pt-3 pb-3 navbar navbar-light"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <span
        className="icon ion-md-home mr-1"
      />
      Home
    </a>
  </nav>,
  <div
    className="container"
  >
    <h2
      className="text-center display-4 mt-5 mb-2"
    >
      Headline
    </h2>
    <div />
    <h2
      className="text-center display-4 mt-5 mb-2"
    >
      List of posts
    </h2>
    <section>
      <article>
        <h3
          className="mt-5"
        />
        <p />
        <a
          href="/post/1"
          onClick={[Function]}
        >
          Read more...
        </a>
      </article>
    </section>
  </div>,
  <footer>
    <div
      className="col-md-offset-3 text-center container"
    >
      <hr
        className="mt-3"
      />
      <p
        className="text-muted small"
      >
        <span>
           Updated 
        </span>
        <time
          className="font-weight-bold"
          data-dateepoch={946684800000}
          data-datetime="2000-01-01T00:00:00.000+00:00"
          title="January 1st 2000, 00:00:00"
        >
          less than 5 seconds ago
        </time>
        <span>
          , built with 
        </span>
        <a
          className="text-muted font-weight-bold"
          href="https://github.com/zeit/next.js"
          onClick={[Function]}
        >
          Next.js
        </a>
        .
      </p>
    </div>
  </footer>,
]
`;
